privacyidea/lib/tokenclass.py

```diff
@@ -926,8 +918,9 @@ def is_pin_change(self, password=False):
         if password:
             key = "next_password_change"
         sdate = self.get_tokeninfo(key)
-        return datetime.datetime.now() > datetime.datetime.strptime(sdate,
-                                                              DATE_FORMAT)
+        #date_change = datetime.datetime.strptime(sdate, DATE_FORMAT)
+        date_change = parse_date_string(sdate)
```

Comment:
The user does never set a date in ``next_pin_change`` manually. This is only achieved by using the tokenclass method. ``set_next_pin_change``.

You are right to use parse_legacy_time here, since it seems it could occur that old timestamps without TZ but with dayfirst are found in the database.